modules/appeals_api/app/models/appeals_api/notice_of_disagreement.rb

```diff
@@ -20,7 +20,8 @@ def self.date_from_string(string)
     attr_encrypted(:form_data, key: Settings.db_encryption_key, marshal: true, marshaler: JsonMarshal::Marshaller)
     attr_encrypted(:auth_headers, key: Settings.db_encryption_key, marshal: true, marshaler: JsonMarshal::Marshaller)
 
-    validate :validate_hearing_type_selection
+    validate :validate_address_unless_homeless, on: :create, if: :form_data_present?
```

Comment:
Please take a look here and make sure I'm not missing any cases. I changed this to `on: create` as opposed to the previous return statement in the custom validation method. After removing PII, the NOD is updated, thus the validation fails unless skipped, since the address will have been removed.